lib/mempool/mempool.js

```diff
@@ -953,6 +967,72 @@ class Mempool extends EventEmitter {
         0);
     }
 
+    // If we reject RBF transactions altogether we can skip these checks,
+    // because incoming conflicts are already rejected as double spends.
+    let conflicts = [];
+    if (this.options.replaceByFee)
+      conflicts = this.getConflicts(tx);
+
+    if (conflicts.length) {
+      let conflictingFees = 0;
+      let totalEvictions = 0;
+
+      // Replacement TXs must pay a higher fee rate than each conflicting TX.
+      // We compare using deltaFee to account for prioritiseTransaction()
+      for (const conflict of conflicts) {
+        conflictingFees += conflict.descFee;
+        totalEvictions += this.countDescendants(conflict) + 1;
+
+        // Replacement TXs must not evict/replace more than 100 descendants
+        if (totalEvictions > 100) {
+          throw new VerifyError(tx,
+          'nonstandard',
+          'too many potential replacements',
+          0);
+        }
+
+        if (entry.getDeltaRate() <= conflict.getDeltaRate()) {
+          throw new VerifyError(tx,
+          'insufficientfee',
+          'insufficient fee',
+          0);
+        }
+      }
+
+      // Replacement TX must also pay for the total fees of all descendant
+      // transactions that will be evicted if an ancestor is replaced.
+      // Thus the replacement "pays for the bandwidth" of all the conflicts.
+      if (entry.deltaFee <= conflictingFees) {
+        throw new VerifyError(tx,
+        'insufficientfee',
+        'insufficient fee',
+        0);
+      }
+
+      // Once the conflicts are all paid for, the replacement TX fee
+      // still needs to cover it's own bandwidth.
+      const feeRemaining = entry.deltaFee - conflictingFees;
+      if (feeRemaining < minFee) {
+        throw new VerifyError(tx,
+          'insufficientfee',
+          'insufficient fee',
+          0);
+      }
+
+      // Replacement TXs can not consume any unconfirmed outputs that were not
```

Comment:
Not request to change, but I used it to better see what was happening, so basically asking if I understood what's up here:

```js
      // Replacement TXs can not consume any unconfirmed outputs that were not
      // already included in the original transactions. They can only add
      // confirmed UTXO, saving the trouble of checking new mempool ancestors.
      // Rule #2 checks
      for (const {prevout} of tx.inputs) {
        // We don't have the transaction in the mempool and it is not an
        // orphan, it means we are spending from the chain, no issues with
        // Rules #2.
        if (!this.map.has(prevout.hash))
          continue;

        // If the prevout is in the mempool and it was already being spent
        // by other transactions (which we are in conflict with) - we
        // are not violating rule #2.
        if (this.isSpent(prevout.hash, prevout.index))
          continue;

        // TX is spending new unconfirmed coin, which violates #2.
        throw new VerifyError(tx,
          'nonstandard',
          'replacement-adds-unconfirmed',
          0);
      }
```